reana_client/cli/workflow.py

```diff
@@ -477,9 +464,65 @@ def workflow_validate(ctx, file):
             err=True)
 
 
+@click.command(
+    'delete',
+    help='Delete a workflow run.')
+@click.option(
+    '-A',
+    '--all',
+    'all_runs',
+    count=True,
```

Comment:
Hmm, the option `--all` may be perhaps misunderstood... Perhaps we could name it more explicitly (e.g. `--all-runs`) or support some wildcard thing such as `-w workflow.*` or `-w workflow.[0-9]*`?

The most frequent functionality that people may want to do and need to understand is (1) deleting runs regarding DB and logs and other such accounting vs (2) deleting runs regarding workspace files and disk space.  It may be easy to misunderstand what "soft" and "hard" do here.  BTW this is also what brought me to muse about `--all`, because the switch could be understood to act on deleting all files and all DB entries and all logs and everything.) 

It may be useful good to think it terms the following thinking framework: *(just an example)* 

command | hide DB entries in CLI? | delete DB entries? | delete logs? | delete workspace? | delete cache?
--- | --- | --- | --- | --- | ---
"soft" | yes  |  no | no | no | yes 
"hard" | no | yes | yes | yes | yes 

A frequent wish may be to remove the files (=to save space) but to keep an entry in the DB (=to see which parameters failed for past runs).  This wish would not be really easily expressible via soff/hard delete dichotomy anymore... It may be more advantageous to offer a CLI like:

```console
$ reana-client delete -w workflow.17 --erase-files --erase-logs
```

with some sane defaults. 

In this way users can compose what they wish to keep and what they wish to delete.  (And `--all` or `--erase-all` would do everything for the given run...)

We can muse further IRL.